xbmc/addons/addoninfo/AddonInfo.cpp

```diff
@@ -243,4 +245,33 @@ const std::string& CAddonInfo::GetTranslatedText(const std::unordered_map<std::s
   return StringUtils::Empty;
 }
 
+std::vector<uint32_t> CAddonInfo::GetKnownInstanceIds() const
+{
+  using namespace XFILE;
+
+  std::vector<uint32_t> ret;
+
+  if (!m_supportInstanceSettings)
+    return ret;
+
+  const std::string searchPath = StringUtils::Format("special://profile/addon_data/{}/", m_id);
+  CFileItemList items;
+  if (!CDirectory::GetDirectory(searchPath, items, ".xml", DIR_FLAG_NO_FILE_DIRS))
+    return ret;
+
+  for (const auto& item : items)
+  {
+    std::string filename = URIUtils::GetFileName(item->GetPath());
+    if (StringUtils::StartsWithNoCase(URIUtils::GetFileName(item->GetPath()), "instance-settings-"))
+    {
+      URIUtils::RemoveExtension(filename);
+      const std::string uid = filename.substr(18);
```

Comment:
please avoid this magic number. you could make "instance-settings-" a variable and work with is string lenght here.